src/inmanta/data/model.py

```diff
@@ -28,12 +28,26 @@
 from inmanta.types import ArgumentTypes, JsonType, SimpleTypes, StrictNonIntBool
 
 
+def validator_timezone_aware_timestamps(value: object) -> object:
+    """
+    A Pydantic validator to ensure that all datetime times are timezone aware.
+    """
+    if isinstance(value, datetime.datetime) and value.tzinfo is None:
+        return value.replace(tzinfo=datetime.timezone.utc)
+    else:
+        return value
+
+
 @stable_api
 class BaseModel(pydantic.BaseModel):
     """
     Base class for all data objects in Inmanta
     """
 
+    _normalize_timestamps: ClassVar[classmethod] = pydantic.validator("*", allow_reuse=True)(
+        validator_timezone_aware_timestamps
+    )
```

Comment:
The problem was that timestamps were not made timezone aware when they are part of a pydantic object. Up to now this was not required, since no pydantic object had a datatime field. The new `LogLine` pydantic object has one. So I extended the conversion to pydantic objects as well. I think it would be good to add this functionality to ISO4 as well.

Reply:
I see, the protocol argument validator only checks the types of the arguments themselves. So we'd have the same behavior for dicts and lists. We don't use datetime for those now but it's an inconsistency in any case.